lib/Utils.vala

```diff
@@ -17,147 +17,234 @@
 
 namespace Gala {
     public class Utils {
-        // Cache xid:pixbuf and icon:pixbuf pairs to provide a faster way aquiring icons
-        static HashTable<string, Gdk.Pixbuf> xid_pixbuf_cache;
-        static HashTable<string, Gdk.Pixbuf> icon_pixbuf_cache;
-        static uint cache_clear_timeout = 0;
+        private class CachedIcon {
```

Comment:
I think a struct or even a `[Compact]` class would be a better fit for this kind of container.